### DISCUSSION TITLE
How to configure the Field / Value selector into the rule builder?

### DISCUSSION THREAD

**Discussion by andreadipalma** [react-querybuilder/react-querybuilder#438] [Q&A]
Hi,
not understanding how to configure a selector to switch between a value or a reference to another field. That is a feature present into the online demo, but (and I'm sorry for that) I'm unable to find a documentation on how to configure it in the builder. At the moment I'm configuring rules based on values OR alternatively rules based on fields, but no chances to dynamically switch between the two.
Very welcome for any support.
Thank you
Andrea

**Comment by jakeboone02** [comment]
If I understand you correctly, you'll be interested in the `valueSource` property of rules, the `valueSources` property of fields, and the `getValueSources` prop on the `QueryBuilder` component.

The most extensive documentation on this feature is here: https://react-querybuilder.js.org/docs/api/querybuilder#getvaluesources

**Reply by andreadipalma** [reply]
Thank you @jakeboone02 , well what I'm seeking is more on the UI part. See the image below. 

![image](https://user-images.githubusercontent.com/8280980/210515852-23283cd7-0c79-432a-93aa-8613c7645248.png)

What I need is how to use the selector (field/value) you put before the value field, to make the rule dynamic. Can you help me with some code snippet for that?
Thank you

**Reply by jakeboone02** [reply]
Sorry, I actually gave you a different link than I intended in the earlier response, but judging by your screenshot it looks like you found it anyway: https://react-querybuilder.js.org/docs/api/valueeditor#value-sources

If either the `getValueSources` prop or the `valueSources` property of the selected field evaluate to an array of `["value", "field"]` or `["field", "value"]`, then the extra selector will be displayed.

If you open the [demo](https://react-querybuilder.js.org/demo) and expand the "Other files" section at the bottom, you'll see `valueSources` in the configuration for the "groupedField#" fields.

**Reply by andreadipalma** [reply]
@jakeboone02 many thanks for the suggestion (and for this beautiful library) now it goes ! 

![image](https://user-images.githubusercontent.com/8280980/210599414-1d22cf64-5b3b-414d-b95f-65d401507811.png)

**Reply by jakeboone02** [reply]
Thanks for the kind words! Glad I could help.
